build.rs

```diff
@@ -267,35 +267,42 @@ fn get_rustc_link_lib(version: &PythonVersion, _: &str, _: bool) -> Result<Strin
 /// 4. `python{major version}.{minor version}`
 ///
 /// If none of the above works, an error is returned
-fn find_interpreter_and_get_config(
-    expected_version: &PythonVersion,
-) -> Result<(PythonVersion, String, Vec<String>), String> {
+fn find_interpreter_and_get_config() -> Result<(PythonVersion, String, Vec<String>), String> {
+    let version = version_from_env();
+
     if let Some(sys_executable) = env::var_os("PYTHON_SYS_EXECUTABLE") {
         let interpreter_path = sys_executable
             .to_str()
             .expect("Unable to get PYTHON_SYS_EXECUTABLE value");
         let (interpreter_version, lines) = get_config_from_interpreter(interpreter_path)?;
-
-        if expected_version == &interpreter_version {
-            return Ok((interpreter_version, interpreter_path.to_owned(), lines));
-        } else {
-            return Err(format!(
+        if version != None && version.as_ref().unwrap() != &interpreter_version {
+            panic!(format!(
```

Comment:
You don't need to use format here. Like println panic uses format internally alrady